framework/quota/src/org/apache/cloudstack/quota/constant/QuotaTypes.java

```diff
@@ -0,0 +1,99 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.cloudstack.quota.constant;
+
+import org.apache.cloudstack.usage.UsageTypes;
+
+import java.util.HashMap;
+
+public class QuotaTypes extends UsageTypes {
+    public static final int CPU_CLOCK_RATE = 15;
+    public static final int CPU_NUMBER = 16;
+    public static final int MEMORY = 17;
+
+    private final Integer quotaType;
+    private final String quotaName;
+    private final String quotaUnit;
+    private final String description;
+    private final String discriminator;
+    private static final HashMap<Integer, QuotaTypes> quotaTypeList = new HashMap<Integer, QuotaTypes>();
+
+    static {
```

Comment:
The ``quotaTypeList`` should be proxied as either a ``Collections.unmodifiableMap`` or built as a ``com.google.collect.UnmodifiableMap`` to ensure immutability and side-effect operation at runtime.